modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -242,68 +240,70 @@ const isDeleteOfLastCharPos = (fromCaption: SugarElement<HTMLTableCaptionElement
         from.isEqual(last) && to.isEqual(first))
   ).getOr(true);
 
-const emptyCaretCaption = (editor: Editor, elm: SugarElement<Node>): Optional<boolean> =>
-  emptyElement(editor, elm);
+const emptyCaretCaption = (editor: Editor, elm: SugarElement<Node>): Optional<Optional<() => void>> =>
```

Comment:
Because it's a tri-state: "Could not produce result", "Result is none", "Result is some", and in the event of the first we run a thunk.

Reply:
To clarify, this was previously `Optional<boolean>`. Since this refactoring has changed this to `Optional<() => void>` there's no longer any difference between what was before `Optional.none` and `Optional<false>` at this particular point in the code. Since the function here runs a thunk on `Optional.none` but not on false, I had to work around it, and this seemed like the one that worked the best with Optionals.

Reply:
I think that could be refactored away. I think the Optional booleans should be able to be reworked into folds seems that the `deleteBetweenCells` and the `deleteCaretInsideCaption` functions are the ones causing this all to be optional booleans. If they are reworked a bit maybe using folds instead of binds then I think it should work. Then `emptyElement` becomes a return of a thunked option of some and the `deleteCaretInsideCaption` would either return a thunked action that some something or a noop thunk to block the default action. The `deleteCaretCells` default action is not to block So that should just return a thunked option if it's some from the `getParentCell` call.

Reply:
I tried to add folds as you wanted, judge for yourself if it improved readability.

Reply:
I think the folds are better here since you should never really have an option in an option. It should probably not have been a optional boolean either it's probably my fault being lazy and just wanting the types to align or something.